include/deal.II/non_matching/mapping_info.h

```diff
@@ -182,79 +390,127 @@ namespace NonMatching
      */
     void
-    compute_mapping_data_for_quadrature(
-      const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-      CellSimilarity::Similarity &cell_similarity,
-      const Quadrature<dim> &     quadrature,
-      MappingData &               mapping_data);
+    resize_data_fields(const unsigned int n_data_point_batches);
 
     /**
-     * Compute the mapping related data for the given @p mapping,
-     * @p cell and @p quadrature that is required by the FEPointEvaluation
-     * class.
+     * Store the unit points.
      */
     void
-    compute_mapping_data_for_immersed_surface_quadrature(
-      const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-      const ImmersedSurfaceQuadrature<dim> &                      quadrature,
-      MappingData &                                               mapping_data);
+    store_unit_points(const unsigned int             unit_points_index_offset,
+                      const unsigned int             n_q_points,
+                      const unsigned int             n_q_points_unvectorized,
+                      const std::vector<Point<dim>> &points);
 
     /**
-     * Compute the mapping related data for the given @p mapping, @p cell,
-     * @p face_no and @p quadrature that is required by the FEPointEvaluation
-     * class.
+     * Store the requested mapping data.
      */
     void
-    compute_mapping_data_for_face_quadrature(
-      const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-      const unsigned int                                          face_no,
-      const Quadrature<dim - 1> &                                 quadrature,
-      MappingData &                                               mapping_data);
+    store_mapping_data(const unsigned int unit_points_index_offset,
+                       const unsigned int n_q_points,
+                       const unsigned int n_q_points_unvectorized,
+                       const MappingData &mapping_data);
+
+    /**
+     * Calculate the compressed cell index.
+     */
+    unsigned int
+    calculate_compressed_cell_index(const unsigned int cell_index) const;
+
+    /**
+     * Calculate the geometry index offset of the current cell/face.
+     */
+    unsigned int
+    calculate_geometry_index_offset(const unsigned int cell_index,
+                                    const unsigned int face_number) const;
+
+    /**
+     * Enum class for reinitialized states.
+     */
+    enum class State
+    {
+      invalid,
+      single_cell,
+      cell_vector,
+      faces_on_cells_in_vector
+    };
+
+    /**
+     * Enum class that stores the currently initialized state
+     * upon the last call of reinit().
+     */
+    State state;
 
     /**
      * The reference points specified at reinit().
+     *
+     * Indexed by @p unit_points_index.
      */
-    std::vector<Point<dim>> unit_points;
+    AlignedVector<Point<dim, VectorizedArrayType>> unit_points;
```

Comment:
I see why you do this. But this change makes the class harder to use if it is used stand-alone.